clients/src/main/java/org/apache/kafka/clients/admin/ConsumerGroupListing.java

```diff
@@ -104,8 +104,8 @@ public boolean equals(Object obj) {
             return false;
         if (isSimpleConsumerGroup != other.isSimpleConsumerGroup)
             return false;
-        if (state == null) {
-            if (other.state != null)
+        if (!state.isPresent()) {
```

Comment:
I think the expectation in the previous code was that `Optional.equals` would handle this case. So can we just get rid of the null check? This depends on the `requireNonNull` in the constructor. Perhaps we should have a unit test to protect it?